tests/Unit/IO/Observers/Test_RegisterEvents.cpp

```diff
@@ -0,0 +1,190 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+#include <type_traits>
+#include <vector>
```

Comment:
`<utility>` for `std::pair`, `std::move`, couldn't find where `<vector>` was used in this test.